master/internal/trial.go

```diff
@@ -352,12 +353,16 @@ func (t *trial) Receive(ctx *actor.Context) error {
 				SlotsNeeded:    slotsNeeded,
 				NonPreemptible: false,
 				Label:          label,
+				ResourcePool:   resourcePool,
 				FittingRequirements: resourcemanagers.FittingRequirements{
 					SingleAgent: false,
 				},
 				TaskActor: ctx.Self(),
 			}
-			ctx.Tell(t.rm, *t.task)
+			if err := ctx.Ask(t.rm, *t.task).Error(); err != nil {
```

Comment:
question: Why is this going from an `ask` to a `tell`? Does adding in a resource pool field mean it can now fail when it couldn't before?

Reply:
It is going from `tell` to `ask` because we need to reject the `TaskRequest` if it doesn't fit in the resource type of any resource pool or the specified resource pool doesn't exist.